packages/webapp/src/components/RepeatCropPlan/index.jsx

```diff
@@ -0,0 +1,378 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useEffect, useRef } from 'react';
+import styles from './styles.module.scss';
+import PropTypes from 'prop-types';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import MultiStepPageTitle from '../PageTitle/MultiStepPageTitle';
+import Form from '../Form';
+import Button from '../Form/Button';
+import Input, { integerOnKeyDown, getInputErrors } from '../Form/Input';
+import RadioGroup from '../Form/RadioGroup';
+import ReactSelect from '../Form/ReactSelect';
+import DaysOfWeekSelect from '../Form/DaysOfWeekSelect';
+import { Label, Main, Error } from '../Typography';
+import { getWeekday, getDate, calculateMonthlyOptions, countOccurrences } from './utils';
+
+export default function PureRepeatCropPlan({
+  cropPlan,
+  farmManagementPlansForCrop,
+  origStartDate,
+  onGoBack = () => {},
+  onContinue = () => {},
+  useHookFormPersist,
+  fromCreation = false,
+}) {
+  const { t } = useTranslation(['translation', 'common']);
+  const {
+    register,
+    handleSubmit,
+    control,
+    getValues,
+    setValue,
+    watch,
+    setError,
+    trigger,
+    formState: { errors, isValid, dirtyFields },
+  } = useForm({
+    mode: 'onChange',
+    shouldUnregister: true,
+    defaultValues: {
+      crop_plan_name: fromCreation ? cropPlan.name : '',
+      plan_start_date: origStartDate,
+      repeat_frequency: 1,
+      repeat_interval: { value: 'week', label: t('REPEAT_PLAN.INTERVAL.WEEK') },
+      days_of_week: [getWeekday(origStartDate)],
+      finish: 'after',
+      after_occurrences: 3,
+    },
+  });
+
+  const { historyCancel } = useHookFormPersist(getValues);
+
+  const intervalOptions = [
+    { value: 'day', label: t('REPEAT_PLAN.INTERVAL.DAY') },
+    { value: 'week', label: t('REPEAT_PLAN.INTERVAL.WEEK') },
+    { value: 'month', label: t('REPEAT_PLAN.INTERVAL.MONTH') },
+    { value: 'year', label: t('REPEAT_PLAN.INTERVAL.YEAR') },
+  ];
+
+  const [monthlyOptions, setMonthlyOptions] = useState([]);
+
+  const totalCount = useRef();
```

Comment:
Why is this ref? I'll just keep that in mind when I work on my ticket.

Reply:
I don't think it will need to be a ref in your view -- I needed the value to not be tied to the render cycle to clear the validation on the "finish on" date component immediately upon interaction with the other inputs; when I kept the count in state the validation function wasn't getting the new value in time.

But it was only for this live feedback on the max count validation; otherwise I don't think it would be necessary.

Reply:
I understand, thank you for the explanation!